### PR TITLE
Remove pre-release process

### PR DESCRIPTION
**Pull request by RogerZhongAWS**
### Motivation
Integration tests now run on all branches

#### Change summary
- removed the need for a pre-release branch
- Fixed a small error in our release ci 

#### Revision diff summary
If there is more than one revision, please explain what has been changed since the last revision.

### Testing
 **Is your change tested? If not, please justify the reason.**  
 **Please list your testing steps and test results.** 
- CI test run result: <link>

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by shangabl** [comment]
LGTM but where do you have a test run from a fork? Maybe merging this from a fork would actually be the best way to prove its working for forks? Wanna make sure the fork CI runs from fork and isn't duplicated.

**Comment by RogerZhongAWS** [comment]
> LGTM but where do you have a test run from a fork? Maybe merging this from a fork would actually be the best way to prove its working for forks? Wanna make sure the fork CI runs from fork and isn't duplicated.

#352
#354

**Review by shangabl — DISMISSED** [review]

**Comment by shangabl** [comment]
Oh I forgot can you update the contributing guidelines?

File: .github/workflows/cppcheck.yml [hunk 1]
```diff
@@ -27,6 +27,9 @@ jobs:
         run: aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws
       - name: Checkout Sources
         uses: actions/checkout@v1
+        with:
```
Comment (marcoemorais-aws) [inline .github/workflows/cppcheck.yml:30]: This is less secure, pls remove.

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -17,18 +17,22 @@ jobs:
...
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.ECR_USER_AWS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.ECR_USER_AWS_KEY_SECRET }}
+          aws-region: us-east-1
+      - name: Login to ECR
+        run: aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws
+      - name: Checkout Sources
+        uses: actions/checkout@v1
+        with:
```
Comment (marcoemorais-aws) [inline .github/workflows/lint.yml:30]: This is less secure, pls remove.

File: .github/workflows/lint.yml [hunk 3]
```diff
@@ -17,18 +17,22 @@ jobs:
...
+          ref: ${{github.event.pull_request.head.ref}}
+          repository: ${{github.event.pull_request.head.repo.full_name}}
+      - name: clang-format script
+        run: |
+          git branch
```
Comment (marcoemorais-aws) [inline .github/workflows/lint.yml:35]: Prolly leftover from your debugging.

**Review by marcoemorais-aws — CHANGES_REQUESTED** [review]

**Review by marcoemorais-aws — APPROVED** [review]

**Review by shangabl — DISMISSED** [review]

**Review by shangabl — APPROVED** [review]

**Review by HarshGandhi-AWS — APPROVED** [review]

**Pull request merged**